### PR TITLE
[Clover step-19] Add role to user

### PR DESCRIPTION
**Pull request by clover-zhao**
### Summary/概要
Description:
Implemented role (admin and user) to user
Changes Made:
- Add role selection for new user create and update
- Add role column to user list page
- Prevent access admin tool for non-login admin users
- Prevent remove the last admin user

How to check the behavior/動作確認
- New or edit user page
    image
- User list page
    image
- Prevent access admin tool for normal users
    image
- Prevent remove last admin user
    image

### CODE REVIEW

**Comment by clover-zhao** [comment]
@hayashiR1 
Could you please review this PR, when you have time?

File: myapp/app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -37,6 +38,11 @@ def show
   end
 
   def destroy
+    if @user.admin? && User.where(role: "admin").count <= 1
+      flash[:alert] = "You cannot delete the only admin user."
```
Comment (hayashiR1) [inline myapp/app/controllers/admin/users_controller.rb]:
```suggestion
    if @user.admin? && User.where(role:'admin').count <= 1
      flash[:alert] = 'You cannot delete the only admin user.'
```
Ruby strings are basically enclosed in single quotes.
(using fablicop is recommended.)

Comment (clover-zhao) [reply myapp/app/controllers/admin/users_controller.rb]: @hayashiR1 Got it! Thanks for your advice.

**Review by hayashiR1 — APPROVED** [review]
dbea51da-26e7-4126-9866-ff8dd5e6b864

**Pull request merged**